### ISSUE TITLE
Early boot DMA protection is disabled in OS despite enabling it in FW 

### ISSUE THREAD

**Issue by johanes2115** [Dasharo/dasharo-issues#548]
### Device

Novacustom nv41 TGL

### Dasharo version

v1.5.0

### Affected component(s) or functionality

After enabling early boot DMA protection in FW and booting into the OS (Ubuntu 22.04), cbmem gives following output:

```
?BM-LOCKDOWN: Enabling boot media protection scheme 'WP_RO only' using CTRL...
?BM-LOCKDOWN: Enabled bootmedia protection
MemoryProtectionCpuArchProtocolNotify:
Boot Policy: DMA protection disabled
?BM-LOCKDOWN: Enabling boot media protection scheme 'WP_RO only' using CTRL...
?BM-LOCKDOWN: Enabled bootmedia protection
?BM-LOCKDOWN: Enabling boot media protection scheme 'WP_RO only' using CTRL...
?BM-LOCKDOWN: Enabled bootmedia protection
MemoryProtectionCpuArchProtocolNotify:
?BM-LOCKDOWN: Skipping enabling boot media protection
MemoryProtectionCpuArchProtocolNotify:
Boot Policy: DMA protection disabled
?BM-LOCKDOWN: Skipping enabling boot media protection
MemoryProtectionCpuArchProtocolNotify:
?BM-LOCKDOWN: Skipping enabling boot media protection
?VT-d PMR HOB not found, not enabling DMA protection
MemoryProtectionCpuArchProtocolNotify:
```

whick suggests, that early boot DMA protection is disabled.

### Brief summary

Early boot DMA protection

### How reproducible

100%

### How to reproduce

1. Power on the device
2. Boot into firmware
3. Go to Dasharo System Features -> Dasharo Security options -> Early Boot DMA Protection
4. Enable the option
5. Save the settings
6. Reset the system and boot into Ubuntu 22.04
7. Open terminal and run `sudo ./cbmem -1 | grep -i protection`
8. Note the results

### Expected behavior

Early boot DMA protection should be enabled

### Actual behavior

Early boot DMA protection is disabled

**Comment by miczyg1** [comment]
@johanes2115 `Boot Policy: DMA protection disabled` says exactly that DMA protection is disabled :)

**Comment by macpijan** [comment]
@johanes2115 What cbmem do we use here?
Isn't the reason the same as we have discussed it was here: https://github.com/Dasharo/dasharo-issues/issues/473

**Comment by mkopec** [comment]
Looks like FSP is not generating a VT-d PMR HOB. It's not configuring VT-d correctly, or just doesn't output the HOB for some reason.

**Comment by johanes2115** [comment]
Output from the newest cbmem looks as follows:
`[ERROR]  VT-d PMR HOB not found, not enabling DMA protection`

**Comment by wessel-novacustom** [comment]
@johanes2115 Does this still happen with the [beta version](https://novacustom.com/forum/d/15-call-for-beta-testers/7)?

**Comment by pietrushnic** [comment]
@wessel-novacustom I doubt he will answer. He no longer works in 3mdeb.

**Comment by macpijan** [comment]
@mkopec will confirm here.

**Comment by miczyg1** [comment]
Indeed, TGL FSP does not produce the required HOB. The HOB was introduced in AlderLake FSP, so any older microarchitecture will not work...

I guess we will have to find a different solution here.

**Comment by mkopec** [comment]
I confirm the issue still happens on v1.5.2

**Comment by mkopec** [comment]
Seems to be fixed on the latest IoT FSP for TigerLake

**Comment by philipanda** [comment]
On Dasharo v1.6.0-rc3 on NV4x TGL and NS5x TGL I get:
```
[INFO ] CBnT: IBB DMA protection: disabled 
[ERROR]  VT-d PMR HOB not found, not enabling DMA protection`
```
despite enabling DMA protection in the setup menu.

Despite the RC using the latest FSP release for TGL: [Edge Platforms TGL-UP3/H IPU 2026.1 (8063_03) FSP](https://github.com/intel/FSP/commit/239368c7d7430f3a5789a307c3c713cd9d364546) (23cf258760b00c4f441776770b1dd233ab709be5)

**Comment by philipanda** [comment]
Switching the FSP type from Client to IoT does not seem to help for NV4x TGL. The results are the same as in the comment above (https://github.com/Dasharo/dasharo-issues/issues/548#issuecomment-4598912578). `cbmem` lists:
- `[ERROR]  VT-d PMR HOB not found, not enabling DMA protection`

Used [Edge Platforms TGL-UP3/H IPU 2026.1 (8063_03) FSP](https://github.com/intel/FSP/commit/239368c7d7430f3a5789a307c3c713cd9d364546) by setting `FSP_TYPE_IOT` in the NovaCustom TGL KConfig which uses Client FSP by default.
